### PR TITLE
[5.1] Template Long Description

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue identified with #42649 .

### Summary of Changes
Complete the code that truncates long descriptions. 
As seen in
administrator/components/com_modules/tmpl/module/edit.php
administrator/components/com_plugins/tmpl/plugin/edit.php
administrator/components/com_scheduler/tmpl/task/edit.php
But only partially seen in
administrator/components/com_templates/tmpl/style/edit.php

each element that used "JGLOBAL_SHOW_FULL_DESCRIPTION", have a parent element with onclick attribute...

### Testing Instructions
Update the descrption of the cassiopeia template in `language\en-GB\tpl_cassiopeia.ini` so that it is over 500 characters long.

### Actual result BEFORE applying this Pull Request
The entire description is displayed

### Expected result AFTER applying this Pull Request
The description is truncated and a new tab is present with the full description

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link
- [x] No documentation changes for manual.joomla.org needed

### Comment
Alternatively the partial support for this truncation functionality can be removed from com_templates

### CODE REVIEW

**Comment by sinahaghparast** [comment]
@brianteeman I saw changed file, thanks...
=========================

now it is better to choose one code style in these four file...

onclick="document.getElementById('myTab').activateTab(document.getElementById('description'));"
or
onclick="document.querySelector('[aria-controls=description]').click();"

**Comment by brianteeman** [comment]
Anyone can submit a pull request

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 650164904889c8ff915eb1a8afad2266b156160d<br /><br />Tested successfully in Joomla 5.1.0-alpha3-dev of 15 January.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42651">issues.joomla.org/tracker/joomla-cms/42651</a>.</sub>

File: administrator/components/com_templates/tmpl/style/edit.php [hunk 1]
```diff
@@ -45,17 +47,31 @@
...
+
+                            if (strlen($truncated) > 500) {
+                                $long_description  = $short_description;
+                                $short_description = HTMLHelper::_('string.truncate', $truncated, 250);
+
+                                if ($short_description == $long_description) {
+                                    $long_description = '';
+                                }
+                            }
+                        }
```
Comment (richard67) [inline administrator/components/com_templates/tmpl/style/edit.php]:
```suggestion
                    $this->fieldset    = 'description';
                    $short_description = Text::_($this->item->xml->description);
                    $long_description  = LayoutHelper::render('joomla.edit.fieldset', $this);

                    if (!$long_description) {
                        $truncated = HTMLHelper::_('string.truncate', $short_description, 550, true, false);

                        if (strlen($truncated) > 500) {
                            $long_description  = $short_description;
                            $short_description = HTMLHelper::_('string.truncate', $truncated, 250);

                            if ($short_description == $long_description) {
                                $long_description = '';
                            }
                        }
                    }
```
Fix PHPCS failing in Drone due to too much indentation.

**Comment by richard67** [comment]
I've restored the previous human test result in the issue tracker as the 2 commits after the test were only a code style change and a clean branch update.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 585dfd6c608ae640958a8b2e9bbdab14365c296d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42651">issues.joomla.org/tracker/joomla-cms/42651</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42651">issues.joomla.org/tracker/joomla-cms/42651</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @brianteeman and also vor testing @toivo and @Quy

**Comment by brianteeman** [comment]
thanks

**Pull request merged**